src/NHibernate.Test/NHSpecificTest/NH1521/Fixture.cs

```diff
@@ -11,7 +11,8 @@ public class Fixture
 	{
 		private static void CheckDialect(Configuration configuration)
 		{
-			if (!configuration.Properties[Environment.Dialect].Contains("MsSql"))
+			var dialect = configuration.Properties[Environment.Dialect];
+			if (!dialect.Contains("MsSql") || dialect.Contains("MsSqlCe"))
```

Comment:
Granted I have not checked it was inheriting from `TestCase`. But well, grabbing the dialect object looks simple: `NHibernate.Dialect.Dialect.GetDialect(configuration.Properties)`.